shm/Player.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Player.h"
+
+#include <numeric>
+
+Player::Player(Ship ship, size_t money)
+    : ship_(std::make_unique<Ship>(ship)), money_(money) {}
+
+size_t Player::getAvailableSpace() const {
+    return available_space_;
+}
+
+size_t Player::getMoney() const {
+    return money_;
+}
+
+void Player::setMoney(size_t amount) {
```

Comment:
just a suggestion, try to keep same names for the variables which set class members
```suggestion
void Player::setMoney(size_t money) {
```